### PR TITLE
Add CUSTOM_OPENCV_URL and CUSTOM_OPENCV_HASH for Arch, openSUSE, and Flatpak

### PR DESCRIPTION
**Pull request by umireon**
AUR, Open Build System, and Flatpak do not allow downloading files in CMake.
Adding CUSTOM_OPENCV variables enables these build systems to download dependent OpenCV files by themselves.

### CODE REVIEW

File: cmake/FetchOpenCV.cmake [hunk 1]
```diff
@@ -1,39 +1,61 @@
...
+    CACHE STRING "Hash of a downloaded OpenCV staitc library tarball")
 
-if(${CMAKE_BUILD_TYPE} STREQUAL Release OR ${CMAKE_BUILD_TYPE} STREQUAL RelWithDebInfo)
-  set(OpenCV_BUILD_TYPE Release)
+if(CUSTOM_OPENCV_URL STREQUAL "")
+  set(USE_PREDEFINED_OPENCV ON)
 else()
-  set(OpenCV_BUILD_TYPE Debug)
+  if(CUSTOM_OPENCV_HASH STREQUAL "")
+    message(FATAL_ERROR "Both of CUSTOM_OPENCV_URL and CUSTOM_OPENCV_MD5 must be present!")
```
Comment (royshil) [inline cmake/FetchOpenCV.cmake]: let's change this message to say `CUSTOM_OPENCV_HASH`

File: cmake/FetchOpenCV.cmake [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
+  else()
+    set(USE_PREDEFINED_OPENCV OFF)
+  endif()
 endif()
 
-if(APPLE)
-  if(OpenCV_BUILD_TYPE STREQUAL Release)
-    set(OpenCV_URL "${OpenCV_BASEURL}/opencv-macos-Release.tar.gz")
-    set(OpenCV_HASH MD5=0875366a03aa44def76ab5a12d3e7b8f)
+if(USE_PREDEFINED_ONNXRUNTIME)
```
Comment (royshil) [inline cmake/FetchOpenCV.cmake]: this should be `USE_PREDEFINED_OPENCV`

**Review by royshil — APPROVED** [review]
fix and ship!

**Comment by umireon** [comment]
I need a new release to ship our latest plugin to AUR, openSUSE, and Flatpak.

File: cmake/FetchOpenCV.cmake (hunk 1 above)
Comment (umireon) [reply cmake/FetchOpenCV.cmake]: Fixed

File: cmake/FetchOpenCV.cmake (hunk 2 above)
Comment (umireon) [reply cmake/FetchOpenCV.cmake]: FIxed

**Pull request merged**